### PR TITLE
[TACHYON-375] Fix the problem that tachyon.master.address is wrong in slaves when using "new TachyonConf()" to create TachyonConf object

### PR DESCRIPTION
**Pull request by Yours-shupeng**
When I use TachyonPerf to test the performance of THCI with 4 nodes, the logs shows every slave using their own hostname as the master address and fail to resolve the true master address. After some trials, I find that the problem is caused by the code "mTachyonConf = new TachyonConf()". When we get TachyonConf object in this way, the default constructor call the loadDefault(true) method in TachyonConf. In the loadDefault(includeSystemProperties : boolean) method, it resolve the local hostname as the default master hostname, and then it load default properties from tachyon-default.properties file. After that, the tachyon.master.address is set using the local hostname and default master port while the master hostname configured by system properties is behind the setting of tachyon.master.address. Therefore, getting TachyonConf object using default constructor " new TachyonConf() " will get the wrong master address in slaves. In this PR, I set the tachyon.master.address after the system properties are loaded.

### CODE REVIEW

**Comment by calvinjia** [comment]
Thanks for the patch! While the change addresses the problem, could it be changed to avoid future problems of this kind as well (ie. with the port)? Also, should site properties be checked as well since the user could have set the address there?

**Comment by Yours-shupeng** [comment]
Thank you for the advice.  I think this problem only exists in tachyon.master.address. Other properties will be overwrited in the order of {default properties, site properties, system properties}. The port will be read from default properties. If property of port is saved in site properties or system properties, the default property will be overwrited.Sorry for that I have mixed the default properties and system properties and I have ignored checking the site properties. Should I define "other properties", get tachyon.master.hostname and tachyon.master.port in the order of {system propertyes, site properties, default properties} ?

**Comment by calvinjia** [comment]
@Yours-shupeng What I meant by port was since `masterAddress` is being constructed via `scheme` + `masterHostname` + `masterPort`, all three of these values should be evaluated in the order you mentioned instead of just default properties. So to calculate `masterAddress` correctly, in the case when the user updates `masterHostname`, `masterPort`, and `useZk` in site or system properties, all three of those values should go through `system` then `site` then `default` properties instead of just `default`.

**Comment by carsonwang** [comment]
From the docs, masterAddress is not expected to be set in site or system properties. Only masterHostName, masterPort and useZk can be set. So why not calculate the masterAddress after defaultProps, siteProps and systemProprs are all combined? We can use the combined mProperties to calculature the correct masterAddress and set it to mProperties.

**Comment by Yours-shupeng** [comment]
@calvinjia @carsonwang  I got it. Thank you for your ideas. I  will fix it and calculate the masterAddress after defaultProps, siteProps and systemProps are all combined.

File: core/src/main/java/tachyon/conf/TachyonConf.java [hunk 1]
```diff
@@ -190,11 +182,19 @@ protected void loadDefault(boolean includeSystemProperties) {
     if (includeSystemProperties) {
       systemProps.putAll(System.getProperties());
     }
-
+
```
Comment (calvinjia) [inline core/src/main/java/tachyon/conf/TachyonConf.java]: Could you remove the trailing whitespace please?

**Comment by haoyuan** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
Aside from the whitespace issue, could you please add a unit test for this? Otherwise LGTM.

**Comment by haoyuan** [comment]
We can’t automatically merge this pull request.

**Comment by Yours-shupeng** [comment]
@haoyuan @calvinjia thanks, I will merge the master and add a unit test.

**Comment by haoyuan** [comment]
We can’t automatically merge this pull request.

**Comment by Yours-shupeng** [comment]
@haoyuan I'm so sorry, I was busy these days, I will fix it today.

**Comment by Yours-shupeng** [comment]
@haoyuan @calvinjia I have merged the master and add a unit test about the masterAddress.

**Comment by calvinjia** [comment]
@Yours-shupeng Could you update to the latest master? LGTM overall

**Comment by calvinjia** [comment]
+1 LGTM

**Pull request merged**